airbyte-integrations/connectors/source-salesforce/unit_tests/api_test.py

```diff
@@ -528,3 +528,59 @@ def test_convert_to_standard_instance(stream_config, stream_api):
     bulk_stream = generate_stream("Account", stream_config, stream_api)
     rest_stream = bulk_stream.get_standard_instance()
     assert isinstance(rest_stream, IncrementalSalesforceStream)
+
+
+def test_bulk_stream_paging(stream_config, stream_api_pk):
+    LastModifiedDate1 = "2022-10-01T00:00:00Z"
+    LastModifiedDate2 = "2022-10-02T00:00:00Z"
+    assert LastModifiedDate1 < LastModifiedDate2
+
+    stream_config["start_date"] = LastModifiedDate1
+    stream: BulkIncrementalSalesforceStream = generate_stream("Account", stream_config, stream_api_pk)
+    stream.page_size = 2
+
+    csv_header = "Field1,LastModifiedDate,Id"
+    pages = [
+        [f"test,{LastModifiedDate1},1", f"test,{LastModifiedDate1},3"],
+        [f"test,{LastModifiedDate1},5", f"test,{LastModifiedDate2},2"],
+        [f"test,{LastModifiedDate2},2", f"test,{LastModifiedDate2},4"],
+        [f"test,{LastModifiedDate2},6"],
+    ]
+
+    with requests_mock.Mocker() as m:
+
+        post_responses = []
+        for job_id, page in enumerate(pages, 1):
+            post_responses.append({"json": {"id": f"{job_id}"}})
+            m.register_uri("GET", stream.path() + f"/{job_id}", json={"state": "JobComplete"})
+            m.register_uri("GET", stream.path() + f"/{job_id}/results", text="\n".join([csv_header] + page))
+            m.register_uri("DELETE", stream.path() + f"/{job_id}")
+        m.register_uri("POST", stream.path(), post_responses)
+
+        records = list(stream.read_records(sync_mode=SyncMode.full_refresh))
+
+        assert records == [
+            {"Field1": "test", "Id": 1, "LastModifiedDate": LastModifiedDate1},
+            {"Field1": "test", "Id": 3, "LastModifiedDate": LastModifiedDate1},
+            {"Field1": "test", "Id": 5, "LastModifiedDate": LastModifiedDate1},
+            {"Field1": "test", "Id": 2, "LastModifiedDate": LastModifiedDate2},
+            {"Field1": "test", "Id": 2, "LastModifiedDate": LastModifiedDate2},  # duplicate record
```

Comment:
Why did you add a duplicate to the test?

Reply:
I just want to show here that duplicates are possible
it's not related to my changes

Reply:
All right, is it something you've already observed with the API?

Reply:
it's possible because SF can store multiple records for the same `cursor field`.
Because we query using `>=` on subsequent queries we can grab duplicates

I mean 2-nd query can produce some result which was on 1-st query

Reply:
Understood, can we deduplicate downstream in the connector or is it too complex? Do you mind opening an issue if you think it's something we can address later?